File: gfg/medium/subset-sum-problem-1611555638/solution.java
class Solution {

    static Boolean isSubsetSum(int arr[], int sum) {
        boolean[] dp = new boolean[sum + 1];

        dp[0] = true;

        for (int num : arr) {
            for (int target = sum; target >= num; target--) {
                if (dp[target - num]) {
                    dp[target] = true;
                }
            }
        }

        return dp[sum];
    }
}

File: gfg/medium/subset-sum-problem-1611555638/solution.py
class Solution:
    def isSubsetSum(self, arr, sum):
        dp = [False] * (sum + 1)
        dp[0] = True

        for num in arr:
            for target in range(sum, num - 1, -1):
                if dp[target - num]:
                    dp[target] = True

        return dp[sum]